### PR TITLE
Go: Initial documentation about interfaces

### PR DESCRIPTION
**Pull request by tobbbles**
Simple documentation on how interfaces in Go should be:

- Located to a `spec.go` file for the package
- Named
- Simpler the better

### CODE REVIEW

File: go/interface_spec.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Interfaces
+
+When providing an interface for a package, it is convention to do this in a `spec.go` file. 
+
+Interfaces follow a naming convention of ending in _'er'_ to describe the behaviour. A `write` package, for example, should fulfil a `Writer` interface. A `scrape` package, should fulfil a `Scraper` interface.
```
Comment (xh3b4sd) [inline go/interface_spec.md]: Except when you have something like `Collector`. 😛

**Review by xh3b4sd — APPROVED** [review]
Please link the new page to the readme like all other pages are linked there.

File: go/interface_spec.md (hunk 1 above)
Comment (tobbbles) [reply go/interface_spec.md]: _/me opens dictionary_

I'll change the wording here, and include a `Collector` example 😅.

Comment (tobbbles) [reply go/interface_spec.md]: Updated 👍

File: README.md [hunk 2]
```diff
@@ -43,6 +43,7 @@
 - [General Style Guide](go/general_style_guide.md)
 - [Idempotency Conventions](go/idempotency_conventions.md)
 - [Imports](go/imports.md)
+- [Interfaces](go/interface_spec.md)
```
Comment (xh3b4sd) [inline README.md]:
```suggestion
- [Interface Spec](go/interface_spec.md)
```

File: go/interface_spec.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Interfaces
```
Comment (xh3b4sd) [inline go/interface_spec.md]:
```suggestion
# Interface Spec
```

**Review by xh3b4sd — APPROVED** [review]
Keeping file name and headings aligned would make my OCD so happy.

**Pull request merged**